sherpa/astro/data.py

```diff
@@ -4892,7 +4892,61 @@ def unsubtract(self):
 
 
 class DataIMG(Data2D):
-    "Image data set, including functions for coordinate transformations"
+    '''Image data set
+
+    This class builds on `sherpa.data.Data2D` to add support for
+    region filters and the ability to describe the data in different coordinate
+    systems, such as the "logical" coordinates (pixels of the image),
+    the "physical" coordinates (e.g. detector coordinates), and the "world"
+    coordinates (e.g. Ra/Dec on the sky).
+    Unlike `sherpa.data.Data2D`, this class does not support sparse data or data
+    with a non-rectangular shape.
+
```

Comment:
I'm not sure this is true. I can create a sparse image

```python
DataIMG("example", [1, 1, 2], [2, 3, 2], [12, 14, 2])
```

There are some things you can't do (without a shape attribute) but you can do some things, like filter the data:

```
sherpa In [1]: load_arrays(1, [1, 1, 2], [2, 3, 2], [12, 14, 2], DataIMG)

sherpa In [2]: image_data()
DataErr: data set '' does not specify a shape

sherpa In [3]: notice2d("circle(1.5, 2.5, 1)")
dataset 1: Field() -> Circle(1.5,2.5,1)

sherpa In [4]: get_dep()
Out[4]: array([12, 14,  2])

sherpa In [5]: get_dep(filter=True)
Out[5]: array([12, 14,  2])

sherpa In [6]: notice2d()
dataset 1: Circle(1.5,2.5,1) -> Field()

sherpa In [7]: notice2d("circle(1.5, 2.5, 0.4)")
dataset 1: Field() -> Circle(1.5,2.5,0.4)

sherpa In [8]: get_dep(filter=True)
Out[8]: array([], dtype=int64)
```

Reply:
Mmm. What is this class good for then? I'll reword to make sure waht it says is actually correct, but all the added functionality is for non-sparse images. I guess it's technically possible to do sparse image, but then it's just like using the parent class.